lighthouse-core/gather/driver.js

```diff
@@ -843,6 +854,10 @@ class Driver {
     this.setNextProtocolTimeout(30 * 1000);
     this.sendCommand('Page.navigate', {url});
 
+    if (waitForNavigated) {
+      await this._waitForFrameNavigated();
```

Comment:
any concerns about racing here? i.e. should we get the promise before `Page.navigate` and then `await` it here?

Reply:
also, any concerns about delaying `_waitForFullyLoaded` until after `Page.frameNavigated` fires?

Reply:
dgozman convinced me that so long as the listener is attached synchronously (which it is inside `this._waitForFrameNavigated`), it is impossible for the event to be received too early. 

Longer explanation was that the earliest possible resolution of the promise would be the next microtask and because the listener is installed before that microtask is executed there's no way to miss the event

Reply:
> dgozman convinced me that so long as the listener is attached synchronously (which it is inside this._waitForFrameNavigated), it is impossible for the event to be received too early.

yeahhhh, good point. As long as no one sticks an `await` in front of that `sendCommand`, I guess?